### ISSUE TITLE
Introduce transmogrifier terminology to solve compiler vs transpiler debate

### ISSUE THREAD

**Issue by mratsim** [nim-lang/website#84]
Discussion started at https://irclogs.nim-lang.org/19-05-2018.html#12:45:21

## Issue

Regularly on various outlets, a heated debate starts whether Nim is compiled or transpiled to C,C++ and Javascript. This detracts the attention of people from Nim features and benefits to a mere question of semantics.

Hacker News:
  - https://news.ycombinator.com/item?id=15161561
  - https://news.ycombinator.com/item?id=16497403

Reddit:
  - https://www.reddit.com/r/programming/comments/7pi2td/this_is_lily_a_programming_language_that_ive_been/dsie1ff/
  - https://www.reddit.com/r/programming/comments/6ak23x/the_future_is_looking_bright_for_python/dhfhyer/
  - https://www.reddit.com/r/golang/comments/384i2s/go_channels_goroutines_and_gc_available_in_nim/crsyq8j/
  - https://www.reddit.com/r/C_Programming/comments/6wy9xm/what_languages_can_be_compiled_to_o_files_and/dmbrvrg/

Github:
  - https://github.com/arnetheduck/nlvm/issues/1#issuecomment-390402510

## Thoughts

Those debates show that the current vocabulary is too limited to describe what Nim does and there is a need to coin another term to end those useless debates.

## Solution

I propose the term `transmogrifier` instead of compiler or transpiler.

> transmogrifier
>
> noun
>
> someone or something that can change or transform its appearance to something else

## Marketing

Here is a short snipper to introduce the term.

> Following heated debates in the Nim community whether Nim was a compiled to C or transpiled to C, and also compiled to JS or transpiled to JS, with arguments one way or the other,
we felt the need to introduce a new terminology as the current ones didn’t address our concerns and failed to communicate what Nim is.
In short Nim transmogrify to C and JS.

**Comment by PMunch** [comment]
> someone or something that can change or transform its _appearance_ to something else

That's the whole issue. Nim doesn't only change the appearance, otherwise none of the compile-time features would work (at least not for JS, I guess it could be theoretically possible to transpile them into some super-complex C macro). The whole reason why Nim is _compiled_ and not _transpiled_ or _transmogrified_ is that it does a non-reversible conversion to something else. You can't look at a piece of Nim generated JS or C code and write the Nim program. In my opinion we should rather write a wiki-page on the website explaining this, so that every time this comes up we could simply link them to that..

**Comment by Araq** [comment]
Sorry, lol, can't take it seriously. We can make it a FAQ entry instead.

**Comment by mratsim** [comment]
It was a joke ;) @dom96 wanted me to add it there :P

**Comment by narimiran** [comment]
> It was a joke

Closing.
